### PR TITLE
Adds the EndermanAttackPlayer event to skript

### PR DESCRIPTION
**Pull request by Phill310**
### Description

This PR adds the `enderman (enrage|anger) event (EndermanAttackPlayerEvent)` to skript with event values `event-player` and `event-entity` (the enderman)

---
**Target Minecraft Versions:** any 
**Requirements:** Paper 
**Related Issues:** #3847

### CODE REVIEW

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 1]
```diff
@@ -678,6 +679,21 @@ public Entity get(LightningStrikeEvent event) {
 				return event.getLightning();
 			}
 		}, 0);
+		// EndermanAttackPlayerEvent
+		if (Skript.classExists("com.destroystokyo.paper.event.entity.EndermanAttackPlayerEvent")) {
+			EventValues.registerEventValue(EndermanAttackPlayerEvent.class, Entity.class, new Getter<Entity, EndermanAttackPlayerEvent>() {
+				@Override
+				public Entity get(EndermanAttackPlayerEvent event) {
+					return event.getEntity();
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:687]: Have you tested this to ensure the event-entity is available and doesn't conflict with the event-player?

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 2]
```diff
@@ -780,6 +780,23 @@ public class SimpleEvents {
...
+						"Note: This does not stop enderman from targeting the player as a result of getting damaged."
+					)
+					.examples(
+						"# Stops endermen from getting angry players with the permission \"safeFrom.enderman\"",
+						"on enderman enrage:",
+							"\tif player has permission \"safeFrom.enderman\":",
+								"\t\tcancel event"
+					)
+					.since("INSERT VERSION")
+					.requiredPlugins("Paper 1.12+");
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:798]:
```suggestion
					.requiredPlugins("Paper");
```

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 3]
```diff
@@ -780,6 +780,23 @@ public class SimpleEvents {
 					.since("INSERT VERSION")
 					.requiredPlugins("Spigot 1.19.4+");
 		}
+
+		if (Skript.classExists("com.destroystokyo.paper.event.entity.EndermanAttackPlayerEvent")) {
+			Skript.registerEvent("Enderman Enrage", SimpleEvent.class, com.destroystokyo.paper.event.entity.EndermanAttackPlayerEvent.class, "enderman (enrage|anger)")
+					.description(
+						"Called when an enderman gets mad because a player looked at them.",
+						"Canceling this event makes the enderman thing the player is wearing a pumpkin.",
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:788]:
```suggestion
```

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java (hunk 1 above)
Comment (Phill310) [reply src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:687]: I was able to set the enderman's name while sending messages to myself so I don't think there is a conflict between the two

**Review by sovdeeth — APPROVED** [review]

**Review by Moderocky — APPROVED** [review]

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 4]
```diff
@@ -710,6 +711,21 @@ public Entity get(LightningStrikeEvent event) {
...
+				}
+			}, EventValues.TIME_NOW);
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:721]: This value shouldn't actually be needed as EndermanAttackPlayerEvent is a type of EntityEvent (for which event-entity is already registered).

```suggestion
```

**Review by APickledWalrus — CHANGES_REQUESTED** [review]

**Review by APickledWalrus — APPROVED** [review]

**Pull request merged**